src/community/generated-geometries/src/test/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategyTest.java

```diff
@@ -333,4 +334,26 @@ public void testThatNonBBoxQueryIsIgnored() throws IOException {
         // then
         assertSame(q.getFilter(), convertedQuery.getFilter());
     }
+    
+ // https://github.com/geosolutions-it/C105-2018-EMSA-VDS/issues/73
+    @Test
+    public void testMultipleBBoxFilterAsORIsConvertedToBetween() {
+        // given
+        final FilterFactory ff = CommonFactoryFinder.getFilterFactory();
+
+        FeatureTypeInfo info = mock(FeatureTypeInfo.class);
+        MetadataMap metadata = getMetadata();
+        when(info.getMetadata()).thenReturn(metadata);
+        Filter bbox1 = ff.bbox("point", -2, -2, 2, 2, "WGS84");
+        Filter bbox2 = ff.bbox("point", -1, -1, 1, 1, "WGS84");
+
+        Filter filter = ff.or(Arrays.asList(bbox1, bbox2));
```

Comment:
I want to see a test where we have also an ``AND`` and where we have a non spatial filter, that non bbox filter should be preserved.

Reply:
tests added for OR, And, Not and also Nested Filters